junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -840,6 +842,25 @@ private static void findNestedClasses(Class<?> clazz, Set<Class<?>> candidates)
 	}
 
 	/**
+	 * Method comparator based upon JUnit4 org.junit.internal.MethodSorter implementation.
+	 */
+	private static int defaultMethodSorter(Method method1, Method method2) {
+		if (method1 == method2) {
+			return 0;
+		}
+		String name1 = method1.getName();
+		String name2 = method2.getName();
+		int comparison = Integer.compare(name1.hashCode(), name2.hashCode());
+		if (comparison == 0) {
+			comparison = name1.compareTo(name2);
+			if (comparison == 0) {
+				comparison = method1.toString().compareTo(method2.toString());
+			}
+		}
+		return comparison;
+	}
```

Comment:
@sormuras Thanks!

I would make an educated guess that the original implementation would be faster than the Stream-like implementation I suggested. How faster? I don't really know.

IMO the Stream-like implementation is easier to read, since it's easier to see which aspects of the methods are chosen for sorting, and in which order those aspects apply. So if performance is no big concern for this method, I'd go for this implementation.

But how do we know if performance is a big problem here or not? Well, it would only be a big performance problem if it was a _bottleneck_ for JUnit 5, which would require profiling to find out. So unless we're willing to profile things now, again I'd go for the Stream-like implementation for clarity.